modules/RTC/TraceablePeerConnection.js

```diff
@@ -1572,14 +1574,19 @@ TraceablePeerConnection.prototype.setLocalDescription = function(
 
     this.trace('setLocalDescription::preTransform', dumpSDP(localSdp));
 
-    if (this.options.preferH264) {
+    if (this.options.disableH264 || this.options.preferH264) {
```

Comment:
having these two settings being discrete looks like it's made things a bit of a pain.  what about a single 'h264' configuration value that could be set to something like 'enabled', 'preferred' or 'disabled'? that way we wouldn't have to worry about all the erroneous interactions between the two settings.

(this could be done as a string in config.js or a value (e.g. 0, 1, 2) either of which would map to a constant or enum in the js itself)

Reply:
TBH I don't like either. I went with this due to the urgency of the matter, but personally I'd like something like `videoCodecs = ['h264', 'vp8']` which would filter all codecs, and only allow those in that order.

Also, adding a new h264 section right now would me more of a pain because of backwards compat, added to the fact that we want this today.

Reply:
ok, i will reluctantly relent :) 

(i didn't mean a new h264 section, btw, just a single value) like:
old:
```
    preferH264: {true|false}
    disableH264: {true|false}
```
new:
```
    h264: {'enabled'|'preferred'|'disabled'}
```

Reply:
One more place where the lack of overridable config parameters makes things harder.